src/databricks/labs/ucx/source_code/notebook.py

```diff
@@ -0,0 +1,331 @@
+from __future__ import annotations  # for type hints
+
+from abc import ABC, abstractmethod
+from ast import parse as parse_python
+from collections.abc import Callable
+from enum import Enum
+
+from sqlglot import ParseError as SQLParseError
+from sqlglot import parse as parse_sql
+from databricks.sdk.service.workspace import Language
+
+NOTEBOOK_HEADER = " Databricks notebook source"
+CELL_SEPARATOR = " COMMAND ----------"
+MAGIC_PREFIX = ' MAGIC'
+LANGUAGE_PREFIX = ' %'
+
+
+class Cell(ABC):
+
+    def __init__(self, source: str):
+        self._original_code = source
+
+    @property
```

Comment:
```suggestion
```
code migration should be an explicit action, properties are implicit actions and preferably immutable

Reply:
that would return the migrated_code once it's generated. I'm not planning to implement its generation here. This property is needed for checking consolidation of the notebook from small bits.